File: src/main/resources/templates/java/solution/src/${packageNameFolder}/BubbleSort.java
package ${packageName};

import java.util.Collections;
import java.util.Date;
import java.util.List;

public class BubbleSort implements SortStrategy {

	public void performSort(List<Date> input) {

		for (int i = input.size() - 1; i >= 0; i--) {
			for (int j = 0; j < i; j++) {
				if (input.get(j).compareTo(input.get(j + 1)) > 0) {
					Date temp = input.get(j);
					input.set(j, input.get(j + 1));
					input.set(j + 1, temp);
				}
			}
		}
	}
}

File: src/main/resources/templates/swift/solution/Sources/swiftExercise/BubbleSort.swift
import Foundation

public class BubbleSort: SortStrategy {
	public func performSort(_ input: inout [Date]) {
        for i in (0 ..< input.count).reversed() {
            for j in 0 ..< i {
                if input[j] > input[j + 1] {
                    let temp: Date! = input[j]
                    input[j] = input[j+1]
                    input[j+1] = temp
                }
            }
        }
    }
}
